src/rollup/index.ts

```diff
@@ -124,357 +103,389 @@ function getInputOptions(rawInputOptions: GenericConfigObject): any {
 }
 
 export default function rollup(
-	rawInputOptions: GenericConfigObject | InputOptions
-): Promise<Bundle | BundleSet> {
+	rawInputOptions: GenericConfigObject
+): Promise<RollupSingleFileBuild | RollupBuild> {
 	try {
 		const inputOptions = getInputOptions(rawInputOptions);
 		initialiseTimers(inputOptions);
 		const graph = new Graph(inputOptions);
 
-		timeStart('BUILD', 1);
+		for (let plugin of graph.plugins) {
+			if (plugin.buildStart) plugin.buildStart.call(graph.pluginContext, inputOptions);
+		}
 
-		const codeSplitting =
-			(inputOptions.experimentalCodeSplitting && typeof inputOptions.input !== 'string') ||
-			inputOptions.experimentalPreserveModules;
+		timeStart('BUILD', 1);
 
-		if (!codeSplitting) {
+		if (!inputOptions.experimentalCodeSplitting) {
```

Comment:
Very nice and clean to have all these validations grouped together. I wonder if it would be possible to move these option validations out of the rollup function and into `getInputOptions`? @